packages/block-editor/src/components/rich-text/event-listeners/paste-handler.js

```diff
@@ -27,12 +27,6 @@ export default ( props ) => ( element ) => {
 			pastePlainText,
 		} = props.current;
 
-		// The event listener is attached to the window, so we need to check if
-		// the target is the element.
```

Comment:
I think this is a valid comment. I probably thought the target would be a focusable element and always the wrapper, but turns out not. Maybe change it to:

```
element.contains( event.target )
```

To make sure we only catch paste within the container.

Reply:
Btw, there's a similar early return in `enter.js`. I wonder if that's causing any bugs?

Reply:
```
! element.contains( event.target )
```

I have confirmed that this approach also works correctly. After a bit of debugging, it appears that `element.target` is sometimes an inner element, rather than the focusable element itself.

> Btw, there's a similar early return in `enter.js`. I wonder if that's causing any bugs?

I've tried debugging close to [this code](https://github.com/WordPress/gutenberg/blob/645c0e7d4156a0b7275d09791f656f65ac183efe/packages/block-editor/src/components/rich-text/event-listeners/enter.js#L25-L29), and as far as I've tested, `element.target` is always the focusable element itself, so it doesn't seem to be causing any kind of bug.

Reply:
Yes, that's super odd.